api.bs

```diff
@@ -1655,23 +1684,49 @@ To perform <dfn>common matching logic</dfn>, given
 
 </div>
 
-#### Last-Touch Attribution #### {#last-touch-attribution}
+#### Last-N-Touch Attribution #### {#last-n-touch-attribution}
 
 <div algorithm>
-To <dfn>fill a histogram with last-touch attribution</dfn>, given a [=set=] of
-  [=impressions=] |matchedImpressions|, an integer |histogramSize|, and an integer |value|:
+To <dfn>fairly allocate credit</dfn>, given a [=list=] of doubles |credit| and an integer |value|:
+1.  Multiply each entry in |credit| by |value|.
+1.  Let |sumCredit| be the sum of |credit|'s [=list/items=].
+1.  Let |normalizedCredit| be the result of element-wise dividing each [=list/item=] in |credit| by |sumCredit|,
+    rounding towards positive Infinity and converting to an integer.
+1.  Let |shuffledIndices| be [=list/get the indices|the indices=] of |credit|, ordered randomly.
```

Comment:
For those that are already integers without rounding, maybe we can consider keeping them as is and only randomly round for non-integers.

Reply:
Oh yes good point. I added a line to remove the integer pieces.